backend/server/routers/collection.py

```diff
@@ -22,130 +22,109 @@
 @router.get("")
 async def get_collections():
     """API to list all collections with details"""
-    try:
-        logger.debug("Listing all the collections...")
-        client = await get_client()
-        collections = await client.aget_collections()
-        if collections is None:
-            return JSONResponse(content={"collections": []})
-        return JSONResponse(
-            content={"collections": [obj.model_dump() for obj in collections]}
-        )
-    except Exception as exp:
-        logger.exception("Failed to get collection")
-        raise HTTPException(status_code=500, detail=str(exp))
+    logger.debug("Listing all the collections...")
+    metadata_store_client = await get_client()
```

Comment:
https://fastapi.tiangolo.com/reference/dependencies/#depends Do you think we can use depend instead of initializing the client everywhere? like `get_collections(metadata_store_client: Depends(get_client))`

Reply:
Nice idea. Will check this out and incorporate if it works

Reply:
I tried adding this but faced issues with the async method. I'll dig deep and pick this up in a new PR. JIRA: TT-9270